ceno_rt/src/lib.rs

```diff
@@ -49,26 +50,37 @@ pub extern "C" fn sys_getenv(_name: *const u8) -> *const u8 {
 #[unsafe(no_mangle)]
 #[linkage = "weak"]
 pub unsafe extern "C" fn sys_rand(recv_buf: *mut u8, words: usize) {
-    unsafe fn step() -> u32 {
```

Comment:
Do you have any insights on:
- the difference in generated assembly code?
- whether using AtomicU32 introduces any additional sys-calls?

Reply:
Since we are working on riscv `zk`vm so one of the concern is the emulator are limited to support native cpu feature in particular for those which is more bind to hardware, such as `lock`, as we need to "prove" lock behaviour with a dedicated zk circuit which is hard to implemented. So I would suggest retain `static mut` behaviours as it doesn't introduce instruction outside of riv32im